### PR TITLE
Fix: Import completion

### PR DESCRIPTION
**Pull request by timotheeguerin**
Fix issue where the import completion would append some or part of the module name creating duplication

For example if picking the `@cadl-lang/compiler` completion

```cadl
import "@|" 
import "@@cadl-lang/compiler"

import "@cadl-lang/|" 
import "@cadl-lang/@cadl-lang/compiler"

import "@cadl|compiler" 
import "@@cadl-lang/compilercompiler"
```

Now all of those will complete to `import "@cadl-lang/compiler"` correctly

### CODE REVIEW

**Comment by nguerrera** [comment]
Could you do the refactoring separately from the fix? It's painful when large chunks of code move around and get bug fixes in the same commit.

**Comment by timotheeguerin** [comment]
> Could you do the refactoring separately from the fix? It's painful when large chunks of code move around and get bug fixes in the same commit.

I'll seperate out the refactoring

**Comment by nguerrera** [comment]
Thanks!

File: packages/compiler/server/serverlib.ts [hunk 1]
```diff
@@ -596,87 +592,6 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
-  /**
```
Comment (timotheeguerin) [inline packages/compiler/server/serverlib.ts:599]: Forgot to remove that one in the refactor PR. There is no change here

File: packages/compiler/server/completion.ts [hunk 2]
```diff
@@ -130,33 +141,30 @@ async function addLibraryImportCompletion(
         program.reportDiagnostic
       );
       if (libPackageJson.cadlMain !== undefined) {
+        const range = {
+          start: file.file.getLineAndCharacterOfPosition(node.pos + 1),
+          end: file.file.getLineAndCharacterOfPosition(node.end - 1),
+        };
         completions.items.push({
+          textEdit: TextEdit.replace(range, dependency),
```
Comment (timotheeguerin) [inline packages/compiler/server/completion.ts:149]: This is basically the important change to this PR

**Review by nguerrera — APPROVED** [review]

**Pull request merged**